Cubical/ZCohomology/EilenbergSteenrodZ.agda

```diff
@@ -157,24 +155,22 @@ fun (suspFunCharac {A = A} n) =
 inv (suspFunCharac {A = A} n) = sMap (suspΩFun (suc n))
 rightInv (suspFunCharac {A = A} n) =
   sElim (λ _ → isOfHLevelPath 2 setTruncIsSet _ _)
-    (λ f → trRec (isProp→isOfHLevelSuc n (setTruncIsSet _ _))
-                  (λ fId → cong ∣_∣₂ (funExt λ x → (cong (ΩKn+1→Kn (suc n))
-                                                            ((λ i → (sym (transportRefl (λ _ → 0ₖ _) i)
-                                                                ∙∙ cong (λ x → suspΩFun (suc n) f x +ₖ 0ₖ _)
-                                                                        (merid x ∙ sym (merid (pt A)))
-                                                                ∙∙ transportRefl (λ _ → 0ₖ _) i))
-                                                         ∙∙ (sym (rUnit (cong (λ x → suspΩFun (suc n) f x +ₖ 0ₖ _)
-                                                                                          ((merid x) ∙ sym (merid (pt A))))))
-                                                         ∙∙ (λ i → congFunct (λ x → rUnitₖ _ (suspΩFun (suc n) f x) i)
-                                                                              (merid x) (sym (merid (pt A))) i)))
-                                                 ∙∙ ΩKn+1→Kn-hom (suc n) (Kn→ΩKn+1 (suc n) (f x))
+        λ f → trRec (isProp→isOfHLevelSuc n (setTruncIsSet _ _))
+                (λ fId → cong ∣_∣₂
+                (funExt (λ x → (λ i → ΩKn+1→Kn (suc n) (sym (rCancel≡refl n i) ∙∙ cong (λ x → suspΩFun (suc n) f x +ₖ 0ₖ _)
+                                                               (merid x ∙ sym (merid (pt A))) ∙∙ rCancel≡refl n i))
+                             ∙∙ (λ i → ΩKn+1→Kn (suc n) (rUnit (cong (λ x → suspΩFun (suc n) f x +ₖ 0ₖ _)
+                                                                       ((merid x) ∙ sym (merid (pt A)))) (~ i)))
+                             ∙∙ (λ i → ΩKn+1→Kn (suc n) (congFunct (λ x → rUnitₖ _ (suspΩFun (suc n) f x) i)
+                                                                     (merid x) (sym (merid (pt A))) i))
+                             ∙∙ ΩKn+1→Kn-hom (suc n) (Kn→ΩKn+1 (suc n) (f x))
```

Comment:
Just a thought, instead of calling `ΩKn+1→Kn` 4 times, maybe you could compose the paths first and then apply it once?

Reply:
Wups, I think I had some type checking issues when rewriting this with the new definition of `rCancelₖ.` Fixed.